Author: Tony Espy <espy@canonical.com>
Subject: NMModem: check for child class' set_mm_enabled func before calling

This patch add a check to NMModem's set_mm_enabled function
to check that a sub-class defines the function before calling
it.  Some modem implementation, such as NMModemOfono, don't
define a set_mm_enabled function, as this control of ofono's
modem 'Online' property is handled externally by urfkill.

Bug: https://bugs.launchpad.net/ubuntu/+source/network-manager/+bug/1445080

---

---
 src/devices/wwan/nm-modem.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: b/src/devices/wwan/nm-modem.c
===================================================================
--- a/src/devices/wwan/nm-modem.c
+++ b/src/devices/wwan/nm-modem.c
@@ -201,7 +201,9 @@ nm_modem_set_mm_enabled (NMModem *self,
 		return;
 	}
 
-	NM_MODEM_GET_CLASS (self)->set_mm_enabled (self, enabled);
+	/* Not all modem classes support set_mm_enabled */
+	if (NM_MODEM_GET_CLASS (self)->set_mm_enabled)
+		NM_MODEM_GET_CLASS (self)->set_mm_enabled (self, enabled);
 
 	/* Pre-empt the state change signal */
 	nm_modem_set_state (self,
